### ISSUE TITLE
using ThemeManifest::TEMPLATES_DIR to work out if a module has "a theme" is too simplistic.

### ISSUE THREAD

**Issue by sunnysideup** [silverstripe/silverstripe-framework#9691]
## Affected Version

SS4

## Description

https://github.com/silverstripe/silverstripe-framework/blob/4/src/View/ThemeManifest.php#L162-L169

We use a rather simple way to decide if a module is "theme-able" - i.e. does it have a templates folder.  I would suggest that we change this to check the composer file and if it is a `silverstripe-module` or a `silverstripe-vendormodule` the include it.  

The reason for this is that in a number of modules I included a "theme-able" css file, but `Requirements::themedCSS` did not work because the module did not have any templates.

## Steps to Reproduce

 - Add a module WITHOUT a `templates` folder.
 - add `Requirements::themedCSS('MyCssFile')` to your code in that module
 - note that this  `MyCssFile` can  not be found, even though it is in the right spot.
 - add `templates` folder to root of your module and it works fine.

**Comment by sminnee** [comment]
Note that in the short term you can fix this by adding this module to the list of active themes for your project. The other approach would be to add an empty `templates/` folder to your module.

We could potentially treat _every_ silverstripe module as a theme as you suggest This will make the list of the themes longer. I doubt that it would break anything, but I would treat it as a feature change.

It feels like an edge-case situation with a clear workaround (as I described). So I think we should only change the behaviour if we're confident it's safe.

**Comment by sunnysideup** [comment]
Hi Sam - thank you for your quick reply.  I dont think it is an edge-case.  I had around 20 modules included with half of them not having a template folder, but half of those having a `Requirements::themedCSS` (I am not using `Requirements::themedJS` - but I think I should use that too). So it is pretty common and it should be as a `Requirements::themedCSS` is a cool way to give a default format and allow devs to create their own interpretation. 

All in all, it it maybe a SS3 hangover as SS4 modules are less likely to use a Requirements::themedCSS I'd say (i.e. less client-side stuff).

**Comment by sminnee** [comment]
* Is the use-case that you build modules that provide new CSS for built-in templates?

 * did this used to work in SS3? If so, was it identical or slightly different?

**Comment by sunnysideup** [comment]
Here is a typical example: module that introduces a form for a form field. In Silverstripe 3 it would work like this:

```php
Requirements::themedCSS('GoogleMapField');
```

In Silverstripe 4 that stopped working because the module was no longer included in the list of theme-able modules. 

On a side note, in SS4, I have moved all CSS files to a `client/css` folder and so I also change my `Requirements::themedCSS` calls to: `Requirements::themedCSS('client/css/MyFile');`; ideally it would just pick that up, but I think that is probably too much magic to add.

**Comment by sminnee** [comment]
Is your form field doing all its HTML generation without a template?  I'm just a bit confused as to how GoogleMapField is created with new CSS but no new HTML templates.

However, if this worked in Silverstripe 3, adding the behaviour back to Silverstripe 4 probably makes sense to me. Do others have concerns with treating all Silverstripe modules of type `silverstripe-module`, `silverstripe-theme`, or `silverstripe-vendormodules` as a theme to be included in the `$default` list? @silverstripe/core-team

**Comment by robbieaverill** [comment]
I think they should all be able to be themed. I'm not sure of the implications of including them in the `$default` list though.

**Comment by sunnysideup** [comment]
> Is your form field doing all its HTML generation without a template? I'm just a bit confused as to how GoogleMapField is created with new CSS but no new HTML templates.

`GoogleMapField` extends TextField - basically adds a bit of JS and CSS to make it work. Sorry, name may not have been perfect, but I am thinking about a FormField where it autocompletes your address.

**Comment by maxime-rainville** [comment]
Why is `Requirements::css()` not an option here?

**Comment by sunnysideup** [comment]
@maxime-rainville yes of course, you can also put it directly into html, but I have a ton of modules using themedCSS and I found they stopped working for the reason listed above.

**Comment by ScopeyNZ** [comment]
I'm heavily in favour of @sunnysideup's suggestion, to base it on the `silverstripe-module` or `silverstripe-vendormodule` in composer. That seems way more preferable than the arbitrary `templates` folder.

**Comment by sminnee** [comment]
Cool; I'd support the change. Don't forget type = silverstripe-theme as well ;-)

**Comment by xini** [comment]
Sorry the dumb question, but what exactly is this trying to solve? We can already load module requirements using
```
Requirements::css('vendor/module: client/css/your-styles.css');
```
What is the idea behind loading these as themed?

**Comment by unclecheese** [comment]
I'm guessing it's because themes cascade, unlike modules.

**Comment by xini** [comment]
@unclecheese thanks Aaron. yeah, I guess that makes sense. But I'd rather combine and amend module requirements in my theme build than having to adhere to the modules' directory structures and override stuff like 
```
Requirements::themedCSS('module1.css');
Requirements::themedCSS('client/css/module2.css');
Requirements::themedCSS('css/module3.css');
Requirements::themedCSS('styles/module4.css');
```
And what about when two modules have their css in `client/css/styles.css`? 
I think this creates more mess than it helps.
